tests/zfs-tests/tests/functional/acl/posix/posix_001_pos.ksh

```diff
@@ -43,19 +44,43 @@
 #
 
 verify_runnable "both"
+
+function cleanup
+{
+	rmdir $TESTDIR/dir.0
+}
+
 log_assert "Verify acltype=posixacl works on file"
+log_onexit cleanup
 
 # Test access to FILE
 log_note "Testing access to FILE"
 log_must touch $TESTDIR/file.0
 log_must setfacl -m g:$ZFS_ACL_STAFF_GROUP:rw $TESTDIR/file.0
-getfacl $TESTDIR/file.0 2> /dev/null | egrep -q "^group:$ZFS_ACL_STAFF_GROUP:rw-$"
+getfacl $TESTDIR/file.0 2> /dev/null | egrep -q \
+    "^group:$ZFS_ACL_STAFF_GROUP:rw-$"
 if [ "$?" -eq "0" ]; then
 	# Should be able to write to file
-	log_must user_run $ZFS_ACL_STAFF1 "echo 'echo test > /dev/null' > $TESTDIR/file.0"
+	log_must user_run $ZFS_ACL_STAFF1 \
+	    "echo 'echo test > /dev/null' > $TESTDIR/file.0"
 
+	# Since $TESTDIR is 777, create a new dir with controlled permissions
+	# for testing that creating a new file is not allowed.
+	log_must mkdir $TESTDIR/dir.0
+	log_must chmod 700 $TESTDIR/dir.0
+	log_must setfacl -m g:$ZFS_ACL_STAFF_GROUP:rw $TESTDIR/dir.0
+	# Confirm permissions
+	ls -l $TESTDIR |grep "dir.0" |grep -q "drwxrw----+"
+	if [ "$?" -ne "0" ]; then
+		log_fail "Permisssions could not be set."
```

Comment:
It would be nice to output the actual value for the logs in case we don't get what's expected. (same on line 80)